src/cryptography/hazmat/asn1/asn1.py

```diff
@@ -31,6 +40,16 @@
 encode_der = declarative_asn1.encode_der
 
 
+def _is_union(field_type: type) -> bool:
+    # NOTE: types.UnionType for `T | U`, typing.Union for `Union[T, U]`.
+    union_types = (
+        (types.UnionType, typing.Union)
+        if hasattr(types, "UnionType")
```

Comment:
Can you add a comment for when we drop the hasattr? (I assume its some new python version)